File: frontend/src/types/facility.ts
export type FacilityType =
  | 'refinery'
  | 'power_plant'
  | 'steel_plant'
  | 'cement_plant'
  | 'lng_terminal';

export interface Facility {
  id: string;
  name: string;
  type: FacilityType;
  latitude: number;
  longitude: number;
  city: string;
  state: string;
  country: string;
}

export const FACILITY_LABELS: Record<FacilityType, string> = {
  refinery: 'Refinery',
  power_plant: 'Power Plant',
  steel_plant: 'Steel Plant',
  cement_plant: 'Cement Plant',
  lng_terminal: 'LNG Terminal',
};

File: backend/app/schemas/facility.py
"""
Facility Pydantic schemas.
Matches the frontend Facility TypeScript interface.
"""
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict


FacilityType = Literal[
    "refinery", "power_plant", "steel_plant", "cement_plant", "lng_terminal"
]


class FacilityResponse(BaseModel):
    """API response schema matching the frontend Facility interface."""

    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    type: FacilityType
    latitude: float
    longitude: float
    city: str
    state: str
    country: str
    source: Optional[str] = "unknown"
